File: week_1/day-3-two-pointer/variations_que/pair_with_target_sum.py
nums = [1, 2, 4, 6, 10]
target = 8

def pairSum(nums, target):
    left, right = 0, len(nums) - 1
    while left < right:
        s = nums[left] + nums[right]
        if s == target:
            return left, right
        elif s < target:
            left += 1
        else:
            right -= 1
    return -1

print(pairSum(nums, target))  # Output: (1, 3)

File: week_1/day-3-two-pointer/variations_que/pair_with_target_sum.js
const nums = [1, 2, 4, 6, 10];
const target = 8;

function pairSum(nums, target) {
    let left = 0, right = nums.length - 1;
    while (left < right) {
        const sum = nums[left] + nums[right];
        if (sum === target) return [left, right];
        sum < target ? left++ : right--;
    }
    return -1;
}

console.log(pairSum(nums, target)); // [1, 3]
